### PR TITLE
Add provides in rubygems to fix build failure

### PR DESCRIPTION
**Pull request by neha170**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Add provides in rubygems to fix build failure

###### Change Log  
- Add provides in rubygems to fix build failure

###### Does this affect the toolchain?  
**NO**

###### Associated issues  
- Build break in pipeline: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=186115&view=results

###### Test Methodology
- Just added provides

### CODE REVIEW

File: SPECS/rubygem-openssl/rubygem-openssl.spec [hunk 1]
```diff
@@ -13,6 +13,8 @@ URL:            https://github.com/ruby/openssl
 Source0:        https://github.com/ruby/openssl/archive/refs/tags/v%{version}.tar.gz#/%{gem_name}-%{version}.tar.gz
 BuildRequires:  git
 BuildRequires:  ruby
+Provides:       openssl = %{version}-%{release}
```
Comment (oliviacrain) [inline SPECS/rubygem-openssl/rubygem-openssl.spec:16]: The provides for the base gem names look incorrect, especially on this one which would override our `openssl` package. Looking at the state of the, should these be `rubygem-%{gem_name}`?

**Review by oliviacrain — CHANGES_REQUESTED** [review]

File: SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec [hunk 2]
```diff
@@ -14,6 +14,8 @@ Source0:        https://github.com/ruby/bigdecimal/archive/refs/tags/v%{version}
 BuildRequires:  git
 BuildRequires:  ruby
 Requires:       ruby(release)
+Provides:       bigdecimal = %{version}-%{release}
+Provides:       rubygem(bigdecimal) = %{version}-%{release}
```
Comment (PawelWMS) [inline SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec:18]: Could you also lint these changes to adjust the spacing?

File: SPECS/rubygem-json/rubygem-json.spec [hunk 3]
```diff
@@ -14,6 +14,8 @@ Source0:        https://github.com/flori/json/archive/refs/tags/v%{version}.tar.
 BuildRequires:  git
 BuildRequires:  ruby
 Requires:       ruby(release)
+Provides:       json = %{version}-%{release}
```
Comment (PawelWMS) [inline SPECS/rubygem-json/rubygem-json.spec:17]: Could we remove these plain `Provides: json` and the like (`bigdecimal`, `io-console`, etc.)? Or replace them with the names we used to have in `ruby.spec`, so `rubygem-bigdecimal`, `rubygem-io-console`, `rubygem-[other]`? The current names seem too generic and I'm worried they might cause issues at some point. I haven't gone through all specs, but it looks like different `(Build)Requires` go for `rubygem([name])` anyway, so we should be fine.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: SPECS/rubygem-json/rubygem-json.spec (hunk 3 above)
Comment (neha170) [reply SPECS/rubygem-json/rubygem-json.spec:17]: Yep, I'll replace them with what we had in ruby.spec before. Thanks

File: SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec [hunk 4]
```diff
@@ -14,6 +14,8 @@ Source0:        https://github.com/ruby/bigdecimal/archive/refs/tags/v%{version}
 BuildRequires:  git
 BuildRequires:  ruby
 Requires:       ruby(release)
+Provides:       rubygem-bigdecimal = %{version}-%{release}
```
Comment (jslobodzian) [inline SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec:17]: Do you need to dash roll these?

File: SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec [hunk 5]
```diff
@@ -14,7 +14,7 @@ Source0:        https://github.com/ruby/bigdecimal/archive/refs/tags/v%{version}
 BuildRequires:  git
 BuildRequires:  ruby
 Requires:       ruby(release)
-Provides:       bigdecimal = %{version}-%{release}
+Provides:       rubygem-bigdecimal = %{version}-%{release}
```
Comment (oliviacrain) [inline SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec:17]: Do we need these `rubygem-%{gem_name}` provides, given that we use that name for the RPM package itself?

Comment (neha170) [reply SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec:17]: Honestly, I am not super sure, but I am adding it to save a potential pipeline failure

Comment (neha170) [reply SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec:17]: Ok, removing the generic ones, sorry for the back and forth

File: SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec (hunk 4 above)
Comment (neha170) [reply SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec:17]: Yep

File: SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec (hunk 5 above)
Comment (PawelWMS) [reply SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec:17]: Ah, you're right, @oliviacrain, I missed the fact that now the packages themselves have the correct names, so only the `rubygem([name])` may be necessary. Or a generator for Ruby packages but that's for a different day.;)

File: SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec (hunk 2 above)
Comment (PawelWMS) [reply SPECS/rubygem-bigdecimal/rubygem-bigdecimal.spec]: My bad, this was correct, I am not sure, why it looked misaligned to me before.

**Review by oliviacrain — APPROVED** [review]

**Review by PawelWMS — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**